File: web/src/services/category-service.ts
import { api } from "./api";
import type { Category, CategoryCreatePayload, CategoryUpdatePayload } from "@/types/category";

export async function getCategories() {
  const response = await api.get("/categories");
  return response.data.data as Category[];
}

export async function getBranchCategories(branchId: number) {
  const response = await api.get(`/categories/branch/${branchId}`);
  return response.data.data as Category[];
}

export async function createCategory(payload: CategoryCreatePayload) {
  const response = await api.post("/categories", payload);
  return response.data.data as Category;
}

export async function updateCategory(categoryId: number, payload: CategoryUpdatePayload) {
  const response = await api.put(`/categories/${categoryId}`, payload);
  return response.data.data as Category;
}

export async function deleteCategory(categoryId: number) {
  const response = await api.delete(`/categories/${categoryId}`);
  return response.data;
}

File: api/app/services/category_service.py
from app.repositories.category_repository import (
    CategoryRepository
)


class CategoryService:

    @staticmethod
    def get_all_categories():
        return CategoryRepository.find_all()

    @staticmethod
    def get_branch_categories(
        branch_id: int
    ):
        return CategoryRepository.find_by_branch_id(
            branch_id
        )

    @staticmethod
    def create_category(payload):
        return CategoryRepository.create(
            payload
        )

    @staticmethod
    def update_category(
        category_id: int,
        payload
    ):
        return CategoryRepository.update(
            category_id,
            payload
        )

    @staticmethod
    def delete_category(
        category_id: int
    ):
        return CategoryRepository.delete(
            category_id
        )